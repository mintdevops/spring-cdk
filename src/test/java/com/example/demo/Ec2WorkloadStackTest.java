
package com.example.demo;

import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.context.properties.EnableConfigurationProperties;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.test.context.ActiveProfiles;
import org.springframework.test.context.ContextConfiguration;
import org.springframework.test.context.junit.jupiter.SpringExtension;

import com.example.demo.config.AppConfig;
import com.example.demo.core.Environment;
import com.example.demo.core.StackFactory;
import com.example.demo.repository.ImageBuilderRepository;
import com.example.demo.repository.ImmutableServerRepository;
import com.example.demo.repository.VpcRepository;
import com.example.demo.service.TaggingService;
import com.example.demo.service.cloudformation.AppStackService;
import com.example.demo.service.cloudformation.ImageStackService;
import com.example.demo.service.cloudformation.StackOutputService;

import au.com.origin.snapshots.Expect;
import au.com.origin.snapshots.junit5.SnapshotExtension;
import au.com.origin.snapshots.serializers.DeterministicJacksonSnapshotSerializer;
import software.amazon.awscdk.core.App;
import software.amazon.awscdk.core.Stack;

@ExtendWith(SnapshotExtension.class)
@ExtendWith(SpringExtension.class)
@SpringBootTest
@EnableConfigurationProperties(value = AppConfig.class)
@ContextConfiguration(classes = {
        AppStackService.class,
        StackFactory.class,
        ImageBuilderRepository.class,
        ImmutableServerRepository.class,
        VpcRepository.class,
        TaggingService.class,
        StackOutputService.class
})
@ActiveProfiles("ec2-workload")
public class Ec2WorkloadStackTest {

    @Autowired
    private AppStackService appStackService;

    @Test
    public void devStack(Expect expect) {
        App app = new App();

        Stack s = appStackService.provision(app, "Dev", Environment.DEV);

        expect.serializer(DeterministicJacksonSnapshotSerializer.class).toMatchSnapshot(
                app.synth()
                   .getStackArtifact(s.getArtifactId())
                   .getTemplate());
    }

    @Test
    public void testStack(Expect expect) {
        App app = new App();

        Stack s = appStackService.provision(app, "Test", Environment.TEST);

        expect.serializer(DeterministicJacksonSnapshotSerializer.class).toMatchSnapshot(
                app.synth()
                   .getStackArtifact(s.getArtifactId())
                   .getTemplate());
    }


    @Test
    public void prodStack(Expect expect) {
        App app = new App();

        Stack s = appStackService.provision(app, "Prod", Environment.PROD);

        expect.serializer(DeterministicJacksonSnapshotSerializer.class).toMatchSnapshot(
                app.synth()
                   .getStackArtifact(s.getArtifactId())
                   .getTemplate());
    }

}
