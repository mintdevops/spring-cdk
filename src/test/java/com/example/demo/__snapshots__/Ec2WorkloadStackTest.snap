com.example.demo.Ec2WorkloadStackTest.devStack=[
  {
    "Mappings": {
      "DevAppAsgAmiMap11447A8C": {
        "eu-west-1": {
          "ami": "app-123"
        }
      }
    },
    "Outputs": {
      "AsgName": {
        "Description": "The AutoScalingGroup Name",
        "Export": {
          "Name": "Dev-DEV-AsgName"
        },
        "Value": {
          "Ref": "DevAppAsgASGD412840B"
        }
      }
    },
    "Resources": {
      "DevAppAppRoleA3E9F5D5": {
        "Properties": {
          "AssumeRolePolicyDocument": {
            "Statement": [
              {
                "Action": "sts:AssumeRole",
                "Effect": "Allow",
                "Principal": {
                  "Service": {
                    "Fn::Join": [
                      "",
                      [
                        "ec2.",
                        {
                          "Ref": "AWS::URLSuffix"
                        }
                      ]
                    ]
                  }
                }
              }
            ],
            "Version": "2012-10-17"
          },
          "ManagedPolicyArns": [
            {
              "Fn::Join": [
                "",
                [
                  "arn:",
                  {
                    "Ref": "AWS::Partition"
                  },
                  ":iam::aws:policy/AmazonSSMManagedInstanceCore"
                ]
              ]
            }
          ],
          "Tags": [
            {
              "Key": "io.mintdevops.app/Owner",
              "Value": "human@mintdevops.io"
            }
          ]
        },
        "Type": "AWS::IAM::Role"
      },
      "DevAppAppSecurityGroupBA17D08E": {
        "Properties": {
          "GroupDescription": "Dev/DevApp/AppSecurityGroup",
          "SecurityGroupEgress": [
            {
              "CidrIp": "0.0.0.0/0",
              "Description": "Allow all outbound traffic by default",
              "IpProtocol": "-1"
            }
          ],
          "SecurityGroupIngress": [
            {
              "CidrIp": "0.0.0.0/0",
              "Description": "from 0.0.0.0/0:8080",
              "FromPort": 8080,
              "IpProtocol": "tcp",
              "ToPort": 8080
            }
          ],
          "Tags": [
            {
              "Key": "io.mintdevops.app/Owner",
              "Value": "human@mintdevops.io"
            }
          ],
          "VpcId": "vpc-12345"
        },
        "Type": "AWS::EC2::SecurityGroup"
      },
      "DevAppAsgASGD412840B": {
        "Properties": {
          "DesiredCapacity": "1",
          "LaunchConfigurationName": {
            "Ref": "DevAppAsgLaunchConfigFC779980"
          },
          "MaxSize": "1",
          "MinSize": "1",
          "Tags": [
            {
              "Key": "io.mintdevops.app/Owner",
              "PropagateAtLaunch": true,
              "Value": "human@mintdevops.io"
            },
            {
              "Key": "Name",
              "PropagateAtLaunch": true,
              "Value": "Dev/DevApp/Asg"
            }
          ],
          "VPCZoneIdentifier": [
            "subnet-456"
          ]
        },
        "Type": "AWS::AutoScaling::AutoScalingGroup",
        "UpdatePolicy": {
          "AutoScalingScheduledAction": {
            "IgnoreUnmodifiedGroupSizeProperties": true
          }
        }
      },
      "DevAppAsgInstanceProfile44C954C2": {
        "Properties": {
          "Roles": [
            {
              "Ref": "DevAppAppRoleA3E9F5D5"
            }
          ]
        },
        "Type": "AWS::IAM::InstanceProfile"
      },
      "DevAppAsgLaunchConfigFC779980": {
        "DependsOn": [
          "DevAppAppRoleA3E9F5D5"
        ],
        "Properties": {
          "IamInstanceProfile": {
            "Ref": "DevAppAsgInstanceProfile44C954C2"
          },
          "ImageId": {
            "Fn::FindInMap": [
              "DevAppAsgAmiMap11447A8C",
              {
                "Ref": "AWS::Region"
              },
              "ami"
            ]
          },
          "InstanceType": "t3.nano",
          "SecurityGroups": [
            {
              "Fn::GetAtt": [
                "DevAppAppSecurityGroupBA17D08E",
                "GroupId"
              ]
            }
          ],
          "UserData": {
            "Fn::Base64": "#!/bin/bash"
          }
        },
        "Type": "AWS::AutoScaling::LaunchConfiguration"
      },
      "SSMParamAsgName4CB79B44": {
        "Properties": {
          "Name": "/Dev-DEV/AsgName",
          "Tags": {
            "io.mintdevops.app/Owner": "human@mintdevops.io"
          },
          "Type": "String",
          "Value": {
            "Ref": "DevAppAsgASGD412840B"
          }
        },
        "Type": "AWS::SSM::Parameter"
      }
    }
  }
]


com.example.demo.Ec2WorkloadStackTest.prodStack=[
  {
    "Mappings": {
      "ProdAppAsgAmiMap660BE5A0": {
        "eu-west-1": {
          "ami": "app-123"
        }
      }
    },
    "Outputs": {
      "AsgName": {
        "Description": "The AutoScalingGroup Name",
        "Export": {
          "Name": "Prod-PROD-AsgName"
        },
        "Value": {
          "Ref": "ProdAppAsgASGBE58625A"
        }
      }
    },
    "Resources": {
      "ProdAppAppRoleFD81F12B": {
        "Properties": {
          "AssumeRolePolicyDocument": {
            "Statement": [
              {
                "Action": "sts:AssumeRole",
                "Effect": "Allow",
                "Principal": {
                  "Service": {
                    "Fn::Join": [
                      "",
                      [
                        "ec2.",
                        {
                          "Ref": "AWS::URLSuffix"
                        }
                      ]
                    ]
                  }
                }
              }
            ],
            "Version": "2012-10-17"
          },
          "ManagedPolicyArns": [
            {
              "Fn::Join": [
                "",
                [
                  "arn:",
                  {
                    "Ref": "AWS::Partition"
                  },
                  ":iam::aws:policy/AmazonSSMManagedInstanceCore"
                ]
              ]
            }
          ],
          "Tags": [
            {
              "Key": "io.mintdevops.app/Owner",
              "Value": "human@mintdevops.io"
            }
          ]
        },
        "Type": "AWS::IAM::Role"
      },
      "ProdAppAppSecurityGroupCAA69C44": {
        "Properties": {
          "GroupDescription": "Prod/ProdApp/AppSecurityGroup",
          "SecurityGroupEgress": [
            {
              "CidrIp": "0.0.0.0/0",
              "Description": "Allow all outbound traffic by default",
              "IpProtocol": "-1"
            }
          ],
          "SecurityGroupIngress": [
            {
              "CidrIp": "0.0.0.0/0",
              "Description": "from 0.0.0.0/0:8080",
              "FromPort": 8080,
              "IpProtocol": "tcp",
              "ToPort": 8080
            }
          ],
          "Tags": [
            {
              "Key": "io.mintdevops.app/Owner",
              "Value": "human@mintdevops.io"
            }
          ],
          "VpcId": "vpc-12345"
        },
        "Type": "AWS::EC2::SecurityGroup"
      },
      "ProdAppAsgASGBE58625A": {
        "Properties": {
          "DesiredCapacity": "1",
          "LaunchConfigurationName": {
            "Ref": "ProdAppAsgLaunchConfig9755B026"
          },
          "MaxSize": "1",
          "MinSize": "1",
          "Tags": [
            {
              "Key": "io.mintdevops.app/Owner",
              "PropagateAtLaunch": true,
              "Value": "human@mintdevops.io"
            },
            {
              "Key": "Name",
              "PropagateAtLaunch": true,
              "Value": "Prod/ProdApp/Asg"
            }
          ],
          "VPCZoneIdentifier": [
            "subnet-456"
          ]
        },
        "Type": "AWS::AutoScaling::AutoScalingGroup",
        "UpdatePolicy": {
          "AutoScalingScheduledAction": {
            "IgnoreUnmodifiedGroupSizeProperties": true
          }
        }
      },
      "ProdAppAsgInstanceProfileA7A02736": {
        "Properties": {
          "Roles": [
            {
              "Ref": "ProdAppAppRoleFD81F12B"
            }
          ]
        },
        "Type": "AWS::IAM::InstanceProfile"
      },
      "ProdAppAsgLaunchConfig9755B026": {
        "DependsOn": [
          "ProdAppAppRoleFD81F12B"
        ],
        "Properties": {
          "IamInstanceProfile": {
            "Ref": "ProdAppAsgInstanceProfileA7A02736"
          },
          "ImageId": {
            "Fn::FindInMap": [
              "ProdAppAsgAmiMap660BE5A0",
              {
                "Ref": "AWS::Region"
              },
              "ami"
            ]
          },
          "InstanceType": "t3.nano",
          "SecurityGroups": [
            {
              "Fn::GetAtt": [
                "GroupId",
                "ProdAppAppSecurityGroupCAA69C44"
              ]
            }
          ],
          "UserData": {
            "Fn::Base64": "#!/bin/bash"
          }
        },
        "Type": "AWS::AutoScaling::LaunchConfiguration"
      },
      "SSMParamAsgName4CB79B44": {
        "Properties": {
          "Name": "/Prod-PROD/AsgName",
          "Tags": {
            "io.mintdevops.app/Owner": "human@mintdevops.io"
          },
          "Type": "String",
          "Value": {
            "Ref": "ProdAppAsgASGBE58625A"
          }
        },
        "Type": "AWS::SSM::Parameter"
      }
    }
  }
]


com.example.demo.Ec2WorkloadStackTest.testStack=[
  {
    "Mappings": {
      "TestAppAsgAmiMap6E94ACF9": {
        "eu-west-1": {
          "ami": "app-123"
        }
      }
    },
    "Outputs": {
      "AsgName": {
        "Description": "The AutoScalingGroup Name",
        "Export": {
          "Name": "Test-TEST-AsgName"
        },
        "Value": {
          "Ref": "TestAppAsgASG61EBEF2B"
        }
      }
    },
    "Resources": {
      "SSMParamAsgName4CB79B44": {
        "Properties": {
          "Name": "/Test-TEST/AsgName",
          "Tags": {
            "io.mintdevops.app/Owner": "human@mintdevops.io"
          },
          "Type": "String",
          "Value": {
            "Ref": "TestAppAsgASG61EBEF2B"
          }
        },
        "Type": "AWS::SSM::Parameter"
      },
      "TestAppAppRole178A5DC8": {
        "Properties": {
          "AssumeRolePolicyDocument": {
            "Statement": [
              {
                "Action": "sts:AssumeRole",
                "Effect": "Allow",
                "Principal": {
                  "Service": {
                    "Fn::Join": [
                      "",
                      [
                        "ec2.",
                        {
                          "Ref": "AWS::URLSuffix"
                        }
                      ]
                    ]
                  }
                }
              }
            ],
            "Version": "2012-10-17"
          },
          "ManagedPolicyArns": [
            {
              "Fn::Join": [
                "",
                [
                  "arn:",
                  {
                    "Ref": "AWS::Partition"
                  },
                  ":iam::aws:policy/AmazonSSMManagedInstanceCore"
                ]
              ]
            }
          ],
          "Tags": [
            {
              "Key": "io.mintdevops.app/Owner",
              "Value": "human@mintdevops.io"
            }
          ]
        },
        "Type": "AWS::IAM::Role"
      },
      "TestAppAppSecurityGroupA4BB64C1": {
        "Properties": {
          "GroupDescription": "Test/TestApp/AppSecurityGroup",
          "SecurityGroupEgress": [
            {
              "CidrIp": "0.0.0.0/0",
              "Description": "Allow all outbound traffic by default",
              "IpProtocol": "-1"
            }
          ],
          "SecurityGroupIngress": [
            {
              "CidrIp": "0.0.0.0/0",
              "Description": "from 0.0.0.0/0:8080",
              "FromPort": 8080,
              "IpProtocol": "tcp",
              "ToPort": 8080
            }
          ],
          "Tags": [
            {
              "Key": "io.mintdevops.app/Owner",
              "Value": "human@mintdevops.io"
            }
          ],
          "VpcId": "vpc-12345"
        },
        "Type": "AWS::EC2::SecurityGroup"
      },
      "TestAppAsgASG61EBEF2B": {
        "Properties": {
          "DesiredCapacity": "1",
          "LaunchConfigurationName": {
            "Ref": "TestAppAsgLaunchConfig3B92DB49"
          },
          "MaxSize": "1",
          "MinSize": "1",
          "Tags": [
            {
              "Key": "io.mintdevops.app/Owner",
              "PropagateAtLaunch": true,
              "Value": "human@mintdevops.io"
            },
            {
              "Key": "Name",
              "PropagateAtLaunch": true,
              "Value": "Test/TestApp/Asg"
            }
          ],
          "VPCZoneIdentifier": [
            "subnet-456"
          ]
        },
        "Type": "AWS::AutoScaling::AutoScalingGroup",
        "UpdatePolicy": {
          "AutoScalingScheduledAction": {
            "IgnoreUnmodifiedGroupSizeProperties": true
          }
        }
      },
      "TestAppAsgInstanceProfile4D5C75D9": {
        "Properties": {
          "Roles": [
            {
              "Ref": "TestAppAppRole178A5DC8"
            }
          ]
        },
        "Type": "AWS::IAM::InstanceProfile"
      },
      "TestAppAsgLaunchConfig3B92DB49": {
        "DependsOn": [
          "TestAppAppRole178A5DC8"
        ],
        "Properties": {
          "IamInstanceProfile": {
            "Ref": "TestAppAsgInstanceProfile4D5C75D9"
          },
          "ImageId": {
            "Fn::FindInMap": [
              "TestAppAsgAmiMap6E94ACF9",
              {
                "Ref": "AWS::Region"
              },
              "ami"
            ]
          },
          "InstanceType": "t3.nano",
          "SecurityGroups": [
            {
              "Fn::GetAtt": [
                "GroupId",
                "TestAppAppSecurityGroupA4BB64C1"
              ]
            }
          ],
          "UserData": {
            "Fn::Base64": "#!/bin/bash"
          }
        },
        "Type": "AWS::AutoScaling::LaunchConfiguration"
      }
    }
  }
]